esi_leap/objects/lease.py

```diff
@@ -208,25 +208,10 @@ def cancel(self, context=None):
             LOG.info('Deleting lease %s', self.uuid)
             try:
                 resource = self.resource_object()
-                if resource.get_lease_uuid() == self.uuid:
-                    resource.remove_lease(self)
-                    if self.parent_lease_uuid is not None:
-                        parent_lease = Lease.get(self.parent_lease_uuid)
-                        resource.set_lease(parent_lease)
+                self.emit_delete_notification(context, resource)
```

Comment:
I'd rename this function, since its main purpose isn't to emit the notification; it's to actually unset the lease, with the notification being a side effect. Maybe something like `deactivate`?